pkg/controllers/reconciler_test.go

```diff
@@ -61,11 +66,19 @@ func TestUpgrade(test *testing.T) {
 			expectedInitimage: initimage,
 			upgrade:           false,
 		},
+		{
+			image:             image + "@sha256:" + preSha256,
+			expectedImage:     image + "@sha256:" + sha256,
+			initimage:         initimage + "@sha256:" + preSha256,
+			expectedInitimage: initimage + "@sha256:" + sha256,
+			upgrade:           true,
+		},
 	}
 
 	for i := range tests {
 		t := tests[i]
-
+		os.Setenv(strings.ReplaceAll(strings.ToUpper(filepath.Base(t.image)), "-", "_")+"_SHA", t.image+"@sha256:"+sha256)
```

Comment:
Umm what would be the way to put the `flag`?
another `bool` type field after `upgrade bool`?

Reply:
I put `isEnvVarSet`. If it's not what you meant, please comment again.!

Reply:
I do not understand what you mean.

Reply:
I think I understood what you mean, and changed.
But, I am not sure if it is 'simpler' though I think it looks 'clearer'.

One thing I am not sure is if it would look better if I make a function for
`strings.ReplaceAll(strings.ToUpper(filepath.Base(name)), "-", "_") + "_SHA"` part?

Reply:
I think having a function for the ReplaceAll-ToUpper combo would make it look better. Especially with the test code. Feel free to make the change.

Reply:
@tkatila Thanks for your comment.!

I was thinking that looks better, but Mikko suggested to "hardcode" that part.
I thought it makes sense and looks intuitive to readers. So, let me just.. use that.